### ISSUE TITLE
Feature request: Add ability to specify number format (e.g. '%.3e') when encoding

### ISSUE THREAD

**Issue by laser101** [g-s-k/matlab-toml#13]
The current implementation uses `num2str()` function to encode double precision numbers to text, but doesn't allow the user to specify a format specifier (`s = num2str(A,formatSpec)`) which can be important for maintaining precision.

This can be added without much effort, so I've created a small [patch]((https://github.com/g-s-k/matlab-toml/files/4752262/matlab-toml-diff.txt)) against master at commit
c7900aa.

The patch allows the user to specify the format specifier when calling `write()` or `encode()` methods. For example...

```
A.one = 1.3;
A.two = 2;
toml.write('out.toml', A, '%.3e');
S = toml.encode(A, '%.3f');
```

Regards,

Russ

**Comment by gaetawoo** [comment]
Point is well taken. I'll look into this. Thank you. I might not get to this in a timely manner for your use though, so if you have your own fork of this with your modifications, that'd would be best to use for now.

**Comment by laser101** [comment]
Yes that's fine...I have patched my local copy and it meets my immediate requirements.

**Comment by RoyiAvital** [comment]
@g-s-k , Will a PR be a step to get this?

**Comment by g-s-k** [comment]
@RoyiAvital If you have time to try to implement this, I would be happy to review a PR 😄

**Comment by RoyiAvital** [comment]
@g-s-k , I will have a look at the patch and try to work something out.
